src/gui/archipelagoWindow.ts

```diff
@@ -0,0 +1,334 @@
+                name: 'server-address',
+                y: y++,
+                text: '',
+                tooltip: 'Enter the server address and port (e.g. archipelago.gg:53055)'
+            }),
+            NewEdit('Slot Name:', {
+                name: 'slot-name',
+                y: y++,
+                text: '',
+                tooltip: 'Enter your slot name'
+            }),
+            NewEdit('Server Password:', {
+                name: 'server-password',
+                y: y++,
+                text: '',
+                tooltip: 'Enter the server password, if there is one'
+            }),
+            [{
+                type: 'button',
+                name: 'cancel-button',
+                x: ww - 90 - 6,
+                y: wh - 6 - 26 - 29,
+                width: 90,
+                height: 26,
+                text: 'Cancel',
+                tooltip: 'Cancels the Archipelago game and set up a local randomizer',
+                onClick: function() {
+                    startGameGui();
+                    settings.rando_archipelago = false;
+                    window.close();
+                }
+            },
+            {
+                type: 'button',
+                name: 'connect-button',
+                x: ww - 90 - 6,
+                y: wh - 6 - 26,
+                width: 90,
+                height: 26,
+                text: 'Connect to Server',
+                onClick: function() {
+                    //TODO: Create function connecting to the Archipelago server
+                    window.close();
+                }
+            },
+            {
+                type: 'button',
+                name: 'start-button',
+                x: ww - 90 - 88 - 6,
+                y: wh - 6 - 26 - 29,
+                width: 85,
+                height: 26,
+                text: 'Start Game!',
+                tooltip: 'Starts your game of Archipelago!',
+                onClick: function() {
+                    onStart();
+                    console.log("At this point, the user should be playing Archipelago! This only needs to be clicked once per multiworld");
+                    window.close();
+                }
+            },
+            ]
+        )
+    });
+    return window;
+}
+
+function archipelagoLocations(){
+    var ww = 700;
+    var wh = 350;
+    let y = 0;
+
+    var Archipelago = GetModule("RCTRArchipelago") as RCTRArchipelago;
+
+    if(!ui.getWindow("archipelago-locations")){
+        var window = ui.openWindow({
+            classification: 'archipelago-locations',
+            title: "Welcome to the Unlock Shop!",
+            width: ww,
+            height: wh,
+            onTabChange: () => {
+                var currentWindow = ui.getWindow("archipelago-locations");
+                if (currentWindow.tabIndex == 0){
+                    currentWindow.findWidget<ListViewWidget>("locked-location-list").items = Archipelago.CreateLockedList();
```

Comment:
could split this line
```suggestion
                    var list = currentWindow.findWidget<ListViewWidget>("locked-location-list");
                    list.items = Archipelago.CreateLockedList();
```